src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -527,8 +527,12 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     for {
       paramMap <- processListClustersParameters(params)
       clusterList <- dbRef.inTransaction { da => da.clusterQuery.listByLabels(paramMap._1, paramMap._2, googleProjectOpt) }
-      visibleClusters <- authProvider.filterUserVisibleClusters(userInfo, clusterList.map(c => (c.googleProject, c.clusterName)).toList)
+      samVisibleClusters <- authProvider.filterUserVisibleClusters(userInfo, clusterList.map(c => (c.googleProject, c.clusterName)).toList)
+      // Making the assumption that users will always be able to access clusters that they create
+      // Fix for https://github.com/DataBiosphere/leonardo/issues/821
+      createdClusters <- dbRef.inTransaction { da => da.clusterQuery.listClustersByCreator(userInfo.userEmail).map(c => c.toList.map(cluster => (cluster.googleProject, cluster.clusterName))) }
```

Comment:
Do we need to make an additional query here? We already have the full list of clusters in the project in the `clusterList` variable. Could you just do:
```
val visibleClusters = samVisibleClusters ::: clusterList.filter(_.creator == userInfo.userEmail)
```